pkg/reconciler/taskrun/taskrun.go

```diff
@@ -258,11 +258,6 @@ func (c *Reconciler) stopSidecars(ctx context.Context, tr *v1beta1.TaskRun) erro
 		return nil
 	}
 
-	// do not continue if the TaskSpec had no sidecars
-	if tr.Status.TaskSpec != nil && len(tr.Status.TaskSpec.Sidecars) == 0 && len(tr.Status.Sidecars) == 0 {
```

Comment:
since the injected sidecars are now populated in the taskrun status, could this be
```
if len(tr.Status.Sidecars) == 0 {
```
?

Reply:
Yes, I can add that back if needed, but I'm sure its providing much value. We check to see if the sidecars are running a few lines down and that would return immediately if there were no sidecars to iterate over so it felt like we were adding unnecessary complexity for negligible performance gains by not checking the TaskRun condition.

Reply:
Alternatively, I think the best path forward might be pushing the check for running sidecars above the TaskRun condition check due to the fact that the state of the TaskRun doesn't matter if there are no sidecars anyways